presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -142,6 +144,19 @@ public static ActualProperties streamBackdoorDeriveProperties(PlanNode node, Lis
         return node.accept(new Visitor(metadata, session), inputProperties);
     }
 
+    public static ActualProperties uniqueToGroupProperties(ActualProperties properties)
+    {
+        if (properties.getLocalProperties().isEmpty()) {
+            return null;
+        }
+        LocalProperty<VariableReferenceExpression> localProperty = Iterables.getOnlyElement(properties.getLocalProperties());
+        if (localProperty instanceof UniqueProperty) {
+            return ActualProperties.builderFrom(properties).local(ImmutableList.of(new GroupingProperty<>(ImmutableList.of(((UniqueProperty<VariableReferenceExpression>) localProperty).getColumn())))).build();
```

Comment:
it seems like this line is the only real thing that this function is doing. Can you explain what's going on here? It looks like it converts a uniqueProperty to a groupingProperty of that unique property. is that right? What's the purpose?